mmv1/third_party/terraform/tpgresource/common_diff_suppress.go.erb

```diff
@@ -179,25 +179,47 @@ func TimestampDiffSuppress(format string) schema.SchemaDiffSuppressFunc {
 	}
 }
 
-// suppress diff when saved is Ipv4 and Ipv6 format while new is required a reference
-// this happens for an internal ip for Private Services Connect
+// Suppresses diff for IPv4 and IPv6 different formats.
+// It also suppress diffs if one IP and one reference are given.
 func InternalIpDiffSuppress(_, old, new string, _ *schema.ResourceData) bool {
-	olds := strings.Split(old, "/")
-	news := strings.Split(new, "/")
-
-	if len(olds) == 2 {
-		if len(news) == 2 {
-			return bytes.Equal(net.ParseIP(olds[0]), net.ParseIP(news[0])) && olds[1] == news[1]
-		} else {
-			return (net.ParseIP(olds[0]) != nil) && (net.ParseIP(new) == nil)
+	addr_netmask_old := strings.Split(old, "/")
+	addr_netmask_new := strings.Split(new, "/")
+
+	// Check if old or new are IPs (with or without netmask)
+	addr_old := func() net.IP {
+		if net.ParseIP(addr_netmask_old[0]) == nil {
+			return net.ParseIP(addr_netmask_old[0])
+		}
+		return net.ParseIP(old)
+	}()
+	addr_new := func() net.IP {
+		if net.ParseIP(addr_netmask_new[0]) == nil {
+			return net.ParseIP(addr_netmask_new[0])
 		}
+		return net.ParseIP(new)
+	}()
+
+	// If old or new are references, return true
+	if (addr_old == nil) || (addr_new == nil) {
+		return true
 	}
 
-	if (net.ParseIP(old) != nil) && (net.ParseIP(new) != nil) {
-		return bytes.Equal(net.ParseIP(old), net.ParseIP(new))
+	addr_equality := false
+	netmask_equality := false
+
+	if bytes.Equal(addr_old, addr_new) {
+		addr_equality = true
+	}
+
+	// If old and new have a netmask compare them,
+	// otherwise always assume they are the same
+	if (len(addr_netmask_old)) == 2 && (len(addr_netmask_new) == 2) {
+		netmask_equality = addr_netmask_old[1] == addr_netmask_new[1]
+	} else {
+		netmask_equality = true
 	}
 
-	return (net.ParseIP(old) != nil) && (net.ParseIP(new) == nil)
```

Comment:
It looks like this says that if the "old" value is a valid IP and the "new" value is _not_ a valid IP, then the diff should be suppressed - and otherwise it should not be.

So for example, if the "old" value is a resource reference, and the new value is a valid IP, I believe this function should return `false`. Same goes for if both are resource references instead of IPs. Those cases don't seem to be captured in the unit tests - could you add them?

Reply:
done